MennaMohie/assignment3/Solution.cpp

```diff
@@ -0,0 +1,172 @@
+#include<iostream>
+#include<string>
+#include<vector>
+#include<map>
+#include<unordered_map>
+#include<set>
+#include"Testing.h" //test macros + overloaded insertion operator for sets
+using namespace std;
+
+enum checkType { wordCheck, prefixCheck };
+
+const int gridCols = 3;
+
+class Trie;
+
+class TrieNode
+{
+	char character;
+	bool terminates = false;
+	unordered_map<char, TrieNode*> children;
+
+	void addWord(string word)
+	{
+		int wordSize = word.length();
+
+		if (wordSize == 0)
+			return;
+
+		char firstChar = word[0];
+		TrieNode *child = children[firstChar];
+		if (child == nullptr)
+		{
+			child = new TrieNode(firstChar);
+			children[firstChar] = child;
+		}
+		if(wordSize>1)
+			child->addWord(word.substr(1, wordSize - 1));
```

Comment:
I think is fine this way, but be aware that this makes a lot of new strings as you go through a word, the other option is to pass around the original word and a index (is more confusing but it's faster).